### ISSUE TITLE
[Examples] Update examples to record total loss via `total_loss += loss.detach()`

### ISSUE THREAD

**Issue by nv-dlasalle** [dmlc/dgl#5302]
## 🚀 Feature
Most of the current examples have a training loops like one of the following:

Using `loss.item()`:
```python3
for x, y in dataloader:
  y_pred = model(x)
  loss = loss_fn(y, y_pred)
  opt.zero_grad()
  model.backward()
  opt.step()
  total_loss += loss.item()
```
Calling `loss.item()` requires synchronizing the CPU and GPU, and prevents overlapping forward/backward/optimizer with the dataloader.

Using `loss` directly:
```python3
for x, y in dataloader:
  y_pred = model(x)
  loss = loss_fn(y, y_pred)
  opt.zero_grad()
  model.backward()
  opt.step()
  total_loss += loss
```
This causes total_loss to be added the computation graph, creating extra work on the GPU.

What we should be doing is calling `loss.detach()`:
```python3
for x, y in dataloader:
  y_pred = model(x)
  loss = loss_fn(y, y_pred)
  opt.zero_grad()
  model.backward()
  opt.step()
  total_loss += loss.detach()
```

This accumulates the loss as a 1-element tensor on the training device, not attached to the computation graph, and allows the CPU to move on to the dataloader when the GPU is still working on forward/backward/optimizer.

**Comment by jermainewang** [comment]
Hi @nv-dlasalle , sorry for the late response. One question. What will happen if the examples print out the total loss every certain iterations? Will that still incur CPU-GPU synchronization?
